libsolidity/ast/AST.cpp

```diff
@@ -136,29 +137,24 @@ FunctionDefinition const* ContractDefinition::receiveFunction() const
 	return nullptr;
 }
 
-vector<EventDefinition const*> const& ContractDefinition::interfaceEvents() const
+vector<EventDefinition const*> const& ContractDefinition::interfaceEvents(bool _ignoreErrors) const
 {
 	return m_interfaceEvents.init([&]{
-		set<string> eventsSeen;
-		vector<EventDefinition const*> interfaceEvents;
-
+		std::set<EventDefinition const*, ASTNode::CompareByID> interfaceEvents;
```

Comment:
In this example,

```
  library L1 {
	  event e();
  }

  library L2 {
	  event e();
  }

  contract C {
	  function f() public {
		  emit L1.e();
		  emit L2.e();
	  }
  }
```

The event `e` would be repeated twice. Technically, only one is enough. Maybe repetition in the abi would be an issue for front ends. Another example would be a contract having the same event as a library.

If you want, we can probably merge this already and potentially fix this issue later.

Reply:
It is two different events. I think it's good to list multiple events. We could filter, but hm.

Reply:
Aren't they the same event from EVM's perspective? There is nothing distinguishing about them, is there?

Reply:
Yeah, maybe you are right. But I would propose to filter them at a different place.